packages/owl-bot/test/scan-googleapis-gen-and-create-pull-requests.ts

```diff
@@ -25,191 +25,195 @@ import path from 'path';
-      },
-      getShortLivedOctokit(): Promise<OctokitType> {
-        return Promise.resolve(octokit as OctokitType);
-      },
-    };
+  create(issue: any) {
+    this.issues.push(issue);
+    issue.html_url = `http://github.com/fake/issues/${this.issues.length}`;
+    return Promise.resolve({data: issue});
   }
+}
 
-  const bYaml: OwlBotYaml = {
-    'deep-copy-regex': [
-      {
-        source: '/b.txt',
-        dest: '/src/b.txt',
+class FakePulls {
+  pulls: any[] = [];
+
+  list() {
+    return Promise.resolve({data: this.pulls});
+  }
+
+  create(pull: any) {
+    this.pulls.push(pull);
+    return Promise.resolve({
+      data: {html_url: `http://github.com/fake/pulls/${this.pulls.length}`},
+    });
+  }
+}
+
+function newFakeOctokit(pulls?: FakePulls, issues?: FakeIssues) {
+  return {
+    pulls: pulls ?? new FakePulls(),
+    issues: issues ?? new FakeIssues(),
+    repos: {
+      get() {
+        return {
+          data: {
+            default_branch: 'main',
+          },
+        };
       },
-    ],
-    'deep-remove-regex': ['/src'],
+    },
   };
+}
 
-  it('does nothing when a pull request already exists', async () => {
-    const octokit = {
-      search: {
-        commits() {
-          return Promise.resolve({data: {total_count: 1}});
+/**
+ * Makes a local destination repo where files will be copied to.
+ */
+function makeDestRepo(yaml: OwlBotYaml): GithubRepo {
+  const destDir = makeRepoWithOwlBotYaml(yaml);
+  const destRepo: GithubRepo = {
+    owner: 'googleapis',
+    repo: 'nodejs-spell-check',
+    getCloneUrl(): string {
+      return destDir;
+    },
+    toString() {
+      return `${this.owner}/${this.repo}`;
+    },
+  };
+  return destRepo;
+}
+
+function makeDestRepoAndConfigsStore(
+  yaml: OwlBotYaml
+): [GithubRepo, ConfigsStore] {
+  const destRepo: GithubRepo = makeDestRepo(yaml);
+
+  const configsStore = new FakeConfigsStore(
+    new Map([
+      [
+        destRepo.toString(),
+        {
+          branchName: 'main',
+          commitHash: '456',
+          installationId: 42,
+          yaml: yaml,
         },
-      },
-    };
+      ],
+    ])
+  );
+  configsStore.githubRepos.set(destRepo.toString(), destRepo);
+  return [destRepo, configsStore];
+}
 
+describe('scanGoogleapisGenAndCreatePullRequests', function () {
+  // These tests use git locally and read and write a lot to the file system,
+  // so a slow file system will slow them down.
+  this.timeout(60000); // 1 minute.
+  const abcRepo = makeAbcRepo();
+  const abcCommits = cmd('git log --format=%H', {cwd: abcRepo})
+    .toString('utf8')
+    .split(/\r?\n/)
+    .map(s => s.trim())
+    .filter(s => s);
+
+  beforeEach(() => {
+    cmd('git checkout main', {cwd: abcRepo});
+  });
+
+  it('does nothing with zero repos affected', async () => {
     assert.strictEqual(
       await scanGoogleapisGenAndCreatePullRequests(
         abcRepo,
-        factory(octokit),
+        {} as OctokitFactory,
```

Comment:
I think we could potentially use `sinon.stub(OctokitFactory)`, if we wanted to assert that octokit is actually executed? I'm surprised the empty object being passed in does cause anything to throw?

Reply:
Octokit is never executed.  If it were executed, this test would fail; that's by design.

`scanGoogleapisGenAndCreatePullRequests()` calls `configsStore.findReposAffectedByFileChanges()` which returns 0 repos.  If no repos are affected, then it has nothing to do, and no reason to call Octokit.